src/GoTrueClient.ts

```diff
@@ -643,11 +650,19 @@ export default class GoTrueClient {
       const { currentSession, expiresAt } = data
       const timeNow = Math.round(Date.now() / 1000)
 
-      if (expiresAt < timeNow) {
+      if (expiresAt < timeNow + EXPIRY_MARGIN) {
         if (this.autoRefreshToken && currentSession.refresh_token) {
           const { error } = await this._callRefreshToken(currentSession.refresh_token)
           if (error) {
             console.log(error.message)
+            if (error.message === 'Failed to fetch') {
```

Comment:
There is a better way to find out if a fetch failed due to a network error - It throws. If we can pass that info all the way back up that would be a more robust way of checking this. For example, Safari seems to have "Load failed" as the error message if a network request fails

Reply:
Another way is to trigger the request when the browser comes online via https://developer.mozilla.org/en-US/docs/Web/API/Window/online_event similar to how we do the visibility change

Reply:
Online event is only good for the devices network connection though.   If the connection to the server is down anywhere (or the server is down/very slow) then you still need a way of retrying.

Reply:
I've switched to a NETWORK_FAILURE constant so this works across browser and retries for any rejected fetches.